sympy/assumptions/ask.py

```diff
@@ -363,6 +363,46 @@ def _extract_all_facts(assump, exprs):
                 facts.add(frozenset(args))
     return CNF(facts)
 
+def _normalize_applied_predicates(expr):
+    # Replace Q.gt(a,b) with Q.lt(b,a)
+    expr = expr.replace(
+        lambda e: hasattr(e, 'function') and e.function == Q.gt,
+        lambda e: Q.lt(e.arguments[1], e.arguments[0])
+    )
+    # Replace Q.ge(a,b) with Q.le(b,a)
+    expr = expr.replace(
+        lambda e: hasattr(e, 'function') and e.function == Q.ge,
+        lambda e: Q.le(e.arguments[1], e.arguments[0])
+    )
+    return expr
+
+def _normalize_relations(expr):
+    def _filter(e):
+        return isinstance(e, (Gt, Ge, Lt, Le, Eq, Ne))
+
+    def _replace(e):
+        a, b = e.args
+        if isinstance(e, Gt):
+            return Q.lt(b, a)
+        elif isinstance(e, Ge):
+            return Q.le(b, a)
+        elif isinstance(e, Lt):
+            return Q.lt(a, b)
+        elif isinstance(e, Le):
+            return Q.le(a, b)
+        elif isinstance(e, Eq):
+            return Q.eq(a, b)
+        elif isinstance(e, Ne):
+            return Q.ne(a, b)
+        return e
+
+    return expr.replace(_filter, _replace)
+
+
+def _normalize_expr(expr):
```

Comment:
Let's add a simple docstring string here which notes what normalizations are done. The important thing is to note the final normalized form, i.e., all inequalities and equalities will be represented using `Q.le`, `Q.lt`, `Q.eq`, and `Q.ne` only.